src/connection.rs

```diff
@@ -105,6 +108,11 @@ impl ConnectionHandler for BrokerRepresentation {
 
         let messenger = Arc::new(Messenger::new(BufStream::new(transport), max_message_size));
         messenger.sync_versions().await?;
+        if let Some(sasl_config) = sasl_config {
```

Comment:
@crepererum I didn't refactor this into a separate `setup` function because I wasn't sure `SaslConfig` should leak into the `Messenger`. Right now `SaslConfig` is known to `transport, connection and ClientBuilder`.

Reply:
Looks good to me. I've just checked: this is the only place where we call `sync_versions` and also the place where we set up SOCKS5 and TLS. So I think this is a good spot to call `sasl_handshake` :+1: